server/routes/orderRoutes.js

```diff
@@ -0,0 +1,53 @@
+const router = require('express').Router();
+const Order = require('../db/models/Order');
+const OrderProduct = require('../db/models/OrderProduct');
+
+router.route('/')
+    .post(async (req, res, next) => {
+        try {
+            let currentCart = await Order.findOrCreate(
+                    {
+                        where: {
+                                userId: req.body.userId,
+                                status: 'inCart'
+                            },
+                    }
+                );
+            currentCart = currentCart[0].dataValues;
+            console.log('CURRENT CART: ', currentCart)
+            const orderLine = await OrderProduct.findAll({
+                where: {
+                    productId: req.body.id,
+                    orderId: currentCart.id
+                }
+            });
+            console.log('ORDER LINE: ', orderLine)
+            
+            if (orderLine[0]) {
+                orderLine[0].dataValues.quantity++;
```

Comment:
This pattern of mutating dataValues and then resaving - is a bit funky - I believe you can:

```javascript
const thing = await Thing.findByPk(1);
thing.prop = 'new thing';

await thing.save();
```